src/burnchains/db.rs

```diff
@@ -147,16 +262,554 @@ impl<'a> BurnchainDBTransaction<'a> {
+                                    &block_commit.block_header_hash, block_commit.block_height, block_commit.vtxindex, reward_cycle, parent_ab_rc, &am, (am.len() as u64) < parent_ab_rc);
+
+                    if (am.len() as u64) < parent_ab_rc {
+                        // child is affirming the parent
+                        am.push(AffirmationMapEntry::PoxAnchorBlockPresent);
+                    }
+
+                    (am, Some(parent_ab_rc))
+                }
+                None => {
+                    let mut parent_am = BurnchainDB::get_affirmation_map(
+                        &self.sql_tx,
+                        parent_metadata.affirmation_id,
+                    )?
+                    .expect("BUG: no affirmation map for parent commit");
+
+                    // parent affirms no anchor blocks
+                    test_debug!("Prepare-phase commit {},{},{} does nothing for reward cycle {}, and it builds on a parent {},{} {} which affirms no anchor block (affirms? {})",
+                                    &block_commit.block_header_hash, block_commit.block_height, block_commit.vtxindex, reward_cycle, block_commit.parent_block_ptr, block_commit.parent_vtxindex, &parent_am, (parent_am.len() as u64) < parent_reward_cycle);
+
+                    if (parent_am.len() as u64) < parent_reward_cycle {
+                        // child is affirming the parent
+                        parent_am.push(AffirmationMapEntry::Nothing);
+                    }
+
+                    (parent_am, None)
+                }
+            };
+
+            let num_affirmed = am.len() as u64;
+            for rc in (num_affirmed + 1)..(reward_cycle + 1) {
+                if BurnchainDB::has_anchor_block(&self.sql_tx, rc)? {
+                    test_debug!(
+                        "Commit {},{},{} skips reward cycle {} with anchor block",
+                        &block_commit.block_header_hash,
+                        block_commit.block_height,
+                        block_commit.vtxindex,
+                        rc
+                    );
+                    am.push(AffirmationMapEntry::PoxAnchorBlockAbsent);
+                } else {
+                    // affirmation weight increases even if there's no decision made, because
+                    // the lack of a decision is still an affirmation of all prior decisions
+                    test_debug!(
+                        "Commit {},{},{} skips reward cycle {} without anchor block",
+                        &block_commit.block_header_hash,
+                        block_commit.block_height,
+                        block_commit.vtxindex,
+                        rc
+                    );
+                    am.push(AffirmationMapEntry::Nothing);
+                }
+            }
+
+            test_debug!(
```

Comment:
I think this would be useful as a debug log (and it has enough data items that it would be better as a k-v style log too).

Reply:
Done